compressor/DynamicRangeCompressor.go

```diff
@@ -0,0 +1,65 @@
+package compressor
+
+//DynamicRangeCompressor can process operation that reduces the volume of loud sounds
+//Treadshold it is a volume-level above which it is necessary to begin compression
+//Attack it is a delay for begin compression after exceeding the threshold
+//release it is a time while compression will work
+//compretionRatio it is a coefficient by which the signal will be multiplied
+type DynamicRangeCompressor struct {
+	threshold       float64
+	attack          int
+	release         int
+	compretionRatio float64
+
+	attackTimer  int
+	releaseTimer int
+
+	isInitialize            bool
+	thresholHasBeenExceeded bool
+}
+
+//NewDynamicRangeCompressor creates new DynamicRangeCompressor to process sounds
+func NewDynamicRangeCompressor(threshold, compretionRatio float64, attack, release int) *DynamicRangeCompressor {
```

Comment:
It's convenient to use a struct instead of obscure parameter lists:
```
compressor.NewDynamicRangeCompressor(compressor.Options{
    Threshold: 0.8,
    CompressionRatio: 0.8,
    Attack: 200,
    Release: 1000,
})
```